targets/XPlatCppSdk/source/code/source/playfab/PlayFabAndroidHttpPlugin.cpp

```diff
@@ -0,0 +1,743 @@
+        assert(jniEnv);
+        if (jniEnv == nullptr)
+        {
+            return false;
+        }
+
+        jobject httpRequestObject = requestTask.impl->GetHttpRequestObject();
+        assert(httpRequestObject);
+        if (httpRequestObject == nullptr)
+        {
+            return false;
+        }
+
+        // Call setMethod
+        {
+            jmethodID methodId = jniEnv->GetMethodID(GetHelper().GetHttpRequestClass(), "setMethod", "(Ljava/lang/String;)V");
+            assert(methodId);
+            if (methodId == nullptr)
+            {
+                return false;
+            }
+
+            jstring urlJstr = jniEnv->NewStringUTF("POST");
+            if (urlJstr == nullptr)
+            {
+                return false;
+            }
+            jniEnv->CallVoidMethod(httpRequestObject, methodId, urlJstr);
+            jniEnv->DeleteLocalRef(urlJstr);
+        }
+
+        // Call SetUrl
+        {
+            auto requestUrl = GetUrl(requestTask);
+
+            jmethodID methodId = jniEnv->GetMethodID(GetHelper().GetHttpRequestClass(), "setUrl", "(Ljava/lang/String;)Z");
+            assert(methodId);
+            if (methodId == nullptr)
+            {
+                return false;
+            }
+
+            jstring urlJstr = jniEnv->NewStringUTF(requestUrl.c_str());
+            if (urlJstr == nullptr)
+            {
+                return false;
+            }
+            jboolean result = jniEnv->CallBooleanMethod(httpRequestObject, methodId, urlJstr);
+            jniEnv->DeleteLocalRef(urlJstr);
+            if (result == JNI_FALSE)
+            {
+                return false;
+            }
+        }
+
+        // Call SetPredefinedHeaders
+        SetPredefinedHeaders(requestTask);
+
+        // Call SetHeader
+        auto headers = requestContainer.GetHeaders();
+        if (!headers.empty())
+        {
+            for (auto const &obj : headers)
+            {
+                if (obj.first.length() != 0 && obj.second.length() != 0) // no empty keys or values in headers
+                {
+                    SetHeader(requestTask, obj.first.c_str(), obj.second.c_str());
+                }
+            }
+        }
+
+        // Call SetBody
+        {
+            auto requestUrl = GetUrl(requestTask);
+
+            jmethodID methodId = jniEnv->GetMethodID(GetHelper().GetHttpRequestClass(), "setBody", "([B)V");
+            assert(methodId);
+            if (methodId == nullptr)
+            {
+                return false;
+            }
+
+            size_t payloadSize = 0;
+            void* payload = nullptr;
+            jbyteArray bodyArray = nullptr;
+
+            if (!GetBinaryPayload(requestTask, payload, payloadSize))
+            {
+                // set string payload if binary wasn't provided
+                std::string requestBody = requestContainer.GetRequestBody();
+                payloadSize = (size_t)requestBody.size();
+
+                if (payloadSize > 0)
```

Comment:
The full contents of the code inside this 'if' statement are almost perfectly duplicated starting at line 527.  We should factor out the differing parts and implement the block once.